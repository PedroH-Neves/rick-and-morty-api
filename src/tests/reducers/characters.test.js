import charactersReducers, { initialState } from '../../redux/reducers/characters';

describe('Characters Reducers', () => {
  it('should return the initial state', () => {
    expect(charactersReducers(undefined, {})).toMatchSnapshot();
  });

  it('should handle Fetch Characters Request', () => {
    expect(charactersReducers(initialState,
      {
        type: 'FETCH_CHARACTERS_REQUEST',
      })).toMatchSnapshot();
  });

  it('should handle Fetch Characters Success', () => {
    expect(charactersReducers(initialState,
      {
        type: 'FETCH_CHARACTERS_SUCCESS',
      })).toMatchSnapshot();
  });

  it('should handle Fetch Characters Failure', () => {
    expect(charactersReducers(initialState,
      {
        type: 'FETCH_CHARACTERS_FAILURE',
      })).toMatchSnapshot();
  });
});
