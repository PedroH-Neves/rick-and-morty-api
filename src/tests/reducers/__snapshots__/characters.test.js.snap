// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters Reducers should handle Fetch Characters Failure 1`] = `
Object {
  "characters": Array [],
  "error": undefined,
  "loading": false,
}
`;

exports[`Characters Reducers should handle Fetch Characters Request 1`] = `
Object {
  "characters": Array [],
  "error": "",
  "loading": true,
}
`;

exports[`Characters Reducers should handle Fetch Characters Success 1`] = `
Object {
  "characters": undefined,
  "error": "",
  "loading": false,
}
`;

exports[`Characters Reducers should return the initial state 1`] = `
Object {
  "characters": Array [],
  "error": "",
  "loading": false,
}
`;
